_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -229,80 +232,124 @@ These two HTML `a` elements have the same [accessible name][] and [context][prog
 </html>
 ```
 
+### Failed
+
+#### Failed Example 1
+
+These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They are visually distinguishable thanks to the relationships conveyed through CSS, but go to different resources.
+
+```html
+<html lang="en">
+	<p>
+		<h2>Contact us:</h2>
+		<a href="https://act-rules.github.io/" style="display:inline-block; background: url(/test-assets/shared/chat.png) 0 / 40px no-repeat; padding: 20px 0 20px 50px;">Contact Us</a>
+		<a href="https://www.w3.org/community/act-r/" style="display:inline-block; background: url(/test-assets/shared/phone.png) 0 / 40px no-repeat; padding: 20px 0 20px 50px; margin-left: 40px;">Contact Us</a>
+	</p>
+</html>
+```
+
 #### Failed Example 2
 
-These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They link to web pages that are similar, but have different information in their content.
+These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], but go to different resources.
 
 ```html
 <html lang="en">
 	<div>
-		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html"
-			>Contact us</a
-		>) and get in touch (
-		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
-			>Contact us</a
-		>)
+		<span style="text-align:center;">Contact us</span>
+		<span style="display:flex; justify-content:space-around;">
+			<img src="/test-assets/shared/chat.png" alt="Chat" style="width:50%;">
+			<img src="/test-assets/shared/phone.png" alt="Phone" style="width:50%;">
+		</span>
+		<span style="display:flex; justify-content:space-around;">
+			<a href="https://act-rules.github.io/">Contact Us</a>
+			<a href="https://www.w3.org/community/act-r/">Contact Us</a>
```

Comment:
Actually, also wondering what happens with browsers that show the link URL when hovering them (e.g., Chrome shows it in the bottom left corner of the window). I do personally use that info sometimes, essentially to disambiguate links before clicking 🤔
But then of course, if the URL is enough to disambiguate links, then they are never really ambiguous to users in general unless the URL itself is non-informative (e.g. `contact1.html` vs `contact2.html`).
Should we have an Accessibility Support note to mention that?

Reply:
That's an interesting point to consider while proceeding with your suggested proposal for expectation: 

> For each pair of links in each target set, either the links go to the same place, or there is no visual information on the page to determine they go to different places.

as "visual information on the page" might be not enough clear and might include also the browser visual experience (e.g. as you said Chrome shows the URL on hover or focus).

We might say something like:
> For each pair of links in each target set, one of the following is true:
- either the links resolve to the [same resource][] or to [equivalent resources](#equivalent-resource); or 
- links resolve to different resources but there is no visual information within the content of the page to predict they go to specific different contents.

With "within the content" we restrict the scope to the visual presentation of the HTML content, preventing people to include visual browsers functionalities that give suggestions/hint.

----

I'm not really concerned about the hover functionality, as it behaves consistently with assistive technologies as well; for example, using VO, if you press ctrl+opt+shift+U you can listen the href attribute. Similarly, if you use VoiceControl, you can say "Move mouse to $number" to display the Chrome "hint" with url info.

On the other hand, how many users are aware about that?

And, even if the URL is different and might help differentiating the destinations, there might be url structures that are more or less complex to be understood, especially for non-expert people; if you think about something like:

* https://demo.com/contact-us
* https://demo.com/about-us

or 

* https://demo.com/en-us/category/client-service/contact-us
* https://demo.com/en-us/category/client-service/about-us

or 

* https://demo.com/en-us/category/client-service/contact-us/?j=432847382974328937428942896&f=438243892748329743829&name=demopage
* https://demo.com/en-us/category/client-service/about-us/?j=432847382974328937428942896&f=438243892748329743829&name=demopage

they are very different in terms of understanding.
I would say that, even if the differences might be perceived and understood by someone using the url, the acc name and visual context within the page are still ambiguous to users in general.

Reply:
Good point. Maybe that should be an assumption (that reading the URL as per hovering in a browser or invoking some shortcut in a screen reader isn't enough to disambiguate as far as WCAG is concerned).
Maybe also strengthening the examples by using URLs that are somewhat confusing. On that direction, I think we can build on top of your last example with query parameters that are a SHA to a customers support ticket or products something like that `<a href="../catalog/product.html?product=432847382974328937428942896">View product</a> <a href="../catalog/product.html?product=438243892748329743829">View product</a>`
That would totally be ambiguous even with the URLs… But could be visually disambiguated by aligned pictures of the products (the same way you did for the chat/phone icons).

Reply:
- updated expectation to every pair of links, as per @Jym77 suggestion;
- added assumption to clarify that browser/AT support to display/announce the URL is not enough to disambiguate links.

TODO:
update some example by using confusing URL

Reply:
Added examples with query parameters to emphasize ambiguity.